src/Typesense/VectorSearchQuery.cs

```diff
@@ -0,0 +1,200 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace Typesense;
+
+/// <summary>
+/// Encapsulates the vector query function utilized by Typesense
+/// Examples:
+/// vec:([0.34, 0.66, 0.12, 0.68], k: 10)
+/// vec:([0.34, 0.66, 0.12, 0.68], k: 10, flat_search_cutoff: 20)
+/// vec:([], id: abcd)
+/// </summary>
+public record VectorQuery
+{
+    /// <summary>
+    /// Interal value of the vector
```

Comment:
Internal value?

Reply:
It is just done to comply with static analysis. Because it does not want array types exposed as they can be mutated by reference. 

Thinking about it, I might just disable the static analysis on that line and just expose the array, since if the user decides to mutate the underlying array by reference they're on their own.

Reply:
I was just thinking it was a typo in the comment.  I can't think of a reason you'd ever mutate the underlying array from outside the class (which is why I had it originally as private set).

Allowing a user to mutate the array in any way breaks checks on init such as validating that only the id or vector can be set.